From b37d0a67b7816d28fe662fa6e549926594d79d14 Mon Sep 17 00:00:00 2001
From: Allan Sandfeld Jensen <allan.jensen@qt.io>
Date: Mon, 20 Oct 2025 14:21:07 +0200
Subject: [PATCH 454/553] Fix alpha value after to-FP16 conversions on AVX2

Pick-to: 6.8
Fixes: QTBUG-141255
Change-Id: I2c11709b3168386a8568b109cad4028dfefb979e
Reviewed-by: Eirik Aavitsland <eirik.aavitsland@qt.io>
(cherry picked from commit de1aff1b8009bf921f3b09357412ebfb45a3b4e5)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/gui/painting/qdrawhelper_avx2.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/painting/qdrawhelper_avx2.cpp b/src/gui/painting/qdrawhelper_avx2.cpp
index 72853be6e97..d7496845197 100644
--- a/src/gui/painting/qdrawhelper_avx2.cpp
+++ b/src/gui/painting/qdrawhelper_avx2.cpp
@@ -1525,7 +1525,7 @@ void QT_FASTCALL storeRGBA16FFromARGB32PM_avx2(uchar *dest, const uint *src, int
             const __m128 vsa = _mm_permute_ps(vsf, _MM_SHUFFLE(3, 3, 3, 3));
             __m128 vsr = _mm_rcp_ps(vsa);
             vsr = _mm_sub_ps(_mm_add_ps(vsr, vsr), _mm_mul_ps(vsr, _mm_mul_ps(vsr, vsa)));
-            vsr = _mm_insert_ps(vsr, _mm_set_ss(1.0f), 0x30);
+            vsr = _mm_insert_ps(vsr, vf, 0x30);
             vsf = _mm_mul_ps(vsf, vsr);
         }
         _mm_storel_epi64((__m128i *)(d + i), _mm_cvtps_ph(vsf, 0));
-- 
2.51.2

